src-tauri/src/cropper/mod.rs

```diff
@@ -40,40 +42,119 @@ fn set_transparency_and_level(win: tauri::WebviewWindow, level: u32) {
     .ok();
 }
 
-fn create_record_button_win(app: &AppHandle) {
-    let primary_monitor = app.primary_monitor().unwrap().unwrap();
-    let scale_factor = primary_monitor.scale_factor();
-    let monitor_size: LogicalSize<f64> = primary_monitor.size().to_logical(scale_factor);
-
-    const RECORD_BUTTON_WIDTH: f64 = 200.0;
-    const RECORD_BUTTON_HEIGHT: f64 = 48.0;
-
-    let mut record_win =
-        WebviewWindowBuilder::new(app, "record", WebviewUrl::App("/record".into()))
-            .title("recorder window")
-            .inner_size(RECORD_BUTTON_WIDTH, RECORD_BUTTON_HEIGHT)
-            .position(
-                (monitor_size.width / 2.0) - (RECORD_BUTTON_WIDTH / 2.0),
-                monitor_size.height - 200.0,
-            )
-            .accept_first_mouse(true)
-            .skip_taskbar(true)
-            .shadow(true)
-            .always_on_top(true)
-            .decorations(false)
-            .resizable(false)
-            .visible(false)
-            .effects(WindowEffectsConfig {
-                #[cfg(target_os = "macos")]
-                effects: vec![WindowEffect::Popover],
+fn tune_record_size_position(app: &AppHandle, target: &scap::Target) {
+    let win_maybe = app.get_webview_window("record");
+    if let Some(win) = win_maybe {
+        const RECORD_BUTTON_WIDTH: f64 = 200.0;
+        const RECORD_BUTTON_HEIGHT: f64 = 48.0;
+        match target {
+            scap::Target::Display(display) => {
+                let monitor_size: LogicalSize<f64> = {
+                    let mode = display.raw_handle.display_mode().unwrap();
+                    LogicalSize::<f64>::new(mode.width() as f64, mode.height() as f64)
+                };
+
                 #[cfg(target_os = "macos")]
-                state: Some(WindowEffectState::Active),
+                {
+                    let position: LogicalPosition<f64> = unsafe {
```

Comment:
Tauri should have built-in constructs for getting the monitor bounds. Let's avoid using the `core-graphics` crate for this otherwise we will have to write separate implementations for Windows and Linux as well.